test/controllers/selectDirector.controller.test.ts

```diff
@@ -124,117 +128,182 @@ describe('SelectDirectorController', () => {
+
+        assert.equal(updatedSession.directorsToSign!.length, 1)
+        assert.equal(updatedSession.directorsToSign![0].id, DIRECTOR_1_ID)
+        assert.equal(updatedSession.directorsToSign![0].name, directorName)
+        assert.equal(updatedSession.directorsToSign![0].email, directorEmail)
       })
 
-      await request(app)
-        .post(SELECT_DIRECTOR_URI)
-        .send(form)
-        .expect(MOVED_TEMPORARILY)
-        .expect('Location', DEFINE_SIGNATORY_INFO_URI)
-    })
+      it('should not store the director details if applicant is not a director', async () => {
+        const form: SelectDirectorFormModel = generateSelectDirectorFormModel(NOT_A_DIRECTOR_ID)
 
-    it('should redirect to select signatories if company is multi director and director is selected', async () => {
-      const form: SelectDirectorFormModel = generateSelectDirectorFormModel('123')
+        when(officerService.getActiveDirectorsForCompany(TOKEN, COMPANY_NUMBER)).thenResolve([
+          { ...generateDirectorDetails(), id: DIRECTOR_1_ID }
+        ])
+        when(validator.validate(deepEqual(form), selectDirectorSchema)).thenReturn(null)
 
-      when(officerService.getActiveDirectorsForCompany(TOKEN, COMPANY_NUMBER)).thenResolve([
-        { ...generateDirectorDetails(), id: '123' },
-        { ...generateDirectorDetails(), id: '456' }
-      ])
-      when(validator.validate(deepEqual(form), selectDirectorSchema)).thenReturn(null)
+        const app = createApp(container => {
+          container.rebind(SessionService).toConstantValue(instance(session))
+          container.rebind(CompanyOfficersService).toConstantValue(instance(officerService))
+          container.rebind(FormValidator).toConstantValue(instance(validator))
+        })
 
-      const app = createApp(container => {
-        container.rebind(SessionService).toConstantValue(instance(session))
-        container.rebind(CompanyOfficersService).toConstantValue(instance(officerService))
-        container.rebind(FormValidator).toConstantValue(instance(validator))
-      })
+        await request(app)
+          .post(SELECT_DIRECTOR_URI)
+          .send(form)
+          .expect(MOVED_TEMPORARILY)
 
-      await request(app)
-        .post(SELECT_DIRECTOR_URI)
-        .send(form)
-        .expect(MOVED_TEMPORARILY)
-        .expect('Location', SELECT_SIGNATORIES_URI)
+        verify(session.setDissolutionSession(anything(), anything())).once()
+
+        const sessionCaptor: ArgCaptor2<Request, DissolutionSession> = capture<Request, DissolutionSession>(session.setDissolutionSession)
+        const updatedSession: DissolutionSession = sessionCaptor.last()[1]
+
+        assert.isEmpty(updatedSession.directorsToSign)
+      })
     })
 
-    it('should redirect to select signatories if company is multi director and director is not selected', async () => {
-      const form: SelectDirectorFormModel = generateSelectDirectorFormModel('other')
+    describe('redirect', () => {
```

Comment:
added some sub-describe blocks to groups tests logically. For example, 4 tests check that the correct URL is being redirected to. 3 tests checks that session is being updated correctly. This may flag a lot of indent changes, but the only two tests I've added are:
* should store the director details and email if applicant is a director
* should not store the director details if applicant is not a director